stubs/tqdm/tqdm/std.pyi

```diff
@@ -0,0 +1,272 @@
+    @staticmethod
+    def status_printer(file: str | io.TextIOWrapper | io.StringIO | None): ...
+    @staticmethod
+    def format_meter(
+        n: float,
+        total: float,
+        elapsed: float,
+        ncols: int | None = ...,
+        prefix: str | None = ...,
+        ascii: bool | str | None = ...,
+        unit: str | None = ...,
+        unit_scale: bool | float | None = ...,
+        rate: float | None = ...,
+        bar_format: str | None = ...,
+        postfix: str | Mapping[object, object] | None = ...,
+        unit_divisor: float | None = ...,
+        initial: float | None = ...,
+        colour: str | None = ...,
+    ) -> str: ...
+    @overload
+    def __init__(
+        self,
+        iterable: Iterable[_T] | None,
+        desc: str | None = ...,
+        total: float | None = ...,
+        leave: bool = ...,
+        file: str | io.TextIOWrapper | io.StringIO | None = ...,
+        ncols: int | None = ...,
+        mininterval: float = ...,
+        maxinterval: float = ...,
+        miniters: float | None = ...,
+        ascii: bool | str | None = ...,
+        disable: bool = ...,
+        unit: str = ...,
+        unit_scale: bool | float = ...,
+        dynamic_ncols: bool = ...,
+        smoothing: float = ...,
+        bar_format: str | None = ...,
+        initial: float = ...,
+        position: int | None = ...,
+        postfix: Mapping[object, object] | str | None = ...,
+        unit_divisor: float = ...,
+        write_bytes: bool | None = ...,
+        lock_args: tuple[bool | None, float | None] | tuple[bool | None] | None = ...,
+        nrows: int | None = ...,
+        colour: str | None = ...,
+        delay: float | None = ...,
+        gui: bool = ...,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self,
+        desc: str | None = ...,
+        total: float | None = ...,
+        leave: bool = ...,
+        file: str | io.TextIOWrapper | io.StringIO | None = ...,
+        ncols: int | None = ...,
+        mininterval: float = ...,
+        maxinterval: float = ...,
+        miniters: float | None = ...,
+        ascii: bool | str | None = ...,
+        disable: bool = ...,
+        unit: str = ...,
+        unit_scale: bool | float = ...,
+        dynamic_ncols: bool = ...,
+        smoothing: float = ...,
+        bar_format: str | None = ...,
+        initial: float = ...,
+        position: int | None = ...,
+        postfix: Mapping[object, object] | str | None = ...,
+        unit_divisor: float = ...,
+        write_bytes: bool | None = ...,
+        lock_args: tuple[bool | None, float | None] | tuple[bool | None] | None = ...,
+        nrows: int | None = ...,
+        colour: str | None = ...,
+        delay: float | None = ...,
+        gui: bool = ...,
+    ) -> None: ...
+    def __new__(cls, *_, **__): ...
+    @classmethod
+    def write(
+        cls, s: str, file: str | io.TextIOWrapper | io.StringIO | None = ..., end: str = ..., nolock: bool = ...
```

Comment:
Again, the only required thing of the `file` parameter here is that the object passed in has a `write()` method:

```suggestion
        cls, s: str, file: SupportsWrite[str] | None = ..., end: str = ..., nolock: bool = ...
```

https://github.com/tqdm/tqdm/blob/4f208e72552c4d916aa4fe6a955349ee8b2ed353/tqdm/std.py#L613-L620